src/main/java/jayne/Jayne.java

```diff
@@ -0,0 +1,52 @@
+package jayne;
+
+import java.util.Objects;
+
+import jayne.task.TaskList;
+
+/**
+ * Represents the main class for the Jayne application.
+ * This class encapsulates the initialization of core components
+ * such as Ui, TaskList, and Storage, and contains the main program loop.
+ */
+public class Jayne {
+    private boolean isEnd = true;
+    private TaskList taskList;
+    private Storage storage;
```

Comment:
Appropriate variable naming, abiding by camelCase and understandable variable names.